File: app/src/main/java/com/mdgd/pokemon/ui/pokemons/dto/FilterData.java
package com.mdgd.pokemon.ui.pokemons.dto;

public class FilterData {
    private final boolean byAttack;
    private final boolean byDefence;
    private final boolean byMovement;

    public FilterData(boolean byAttack, boolean byDefence, boolean byMovement) {
        this.byAttack = byAttack;
        this.byDefence = byDefence;
        this.byMovement = byMovement;
    }
}

File: app/src/main/java/com/mdgd/pokemon/ui/pokemons/infra/FilterData.kt
package com.mdgd.pokemon.ui.pokemons.infra;

import java.util.LinkedList;
import java.util.List;

public class FilterData {
    public static final String FILTER_ATTACK = "attack";
    public static final String FILTER_DEFENCE = "defence";
    public static final String FILTER_SPEED = "speed";

    private final List<String> filters;


    public FilterData() {
        this.filters = new LinkedList<>();
    }

    public FilterData(List<String> filters) {
        this.filters = filters;
    }

    public boolean isEmpty() {
        return filters.isEmpty();
    }

    public List<String> getFilters() {
        return filters;
    }
}
